src/main/java/io/vertx/ext/couchdb/impl/CouchdbClientImpl.java

```diff
@@ -5,74 +5,67 @@
 import io.vertx.core.Vertx;
 import io.vertx.core.buffer.Buffer;
 import io.vertx.core.http.HttpMethod;
+import io.vertx.core.internal.VertxInternal;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
-import io.vertx.ext.auth.authentication.Credentials;
 import io.vertx.ext.couchdb.CouchdbClient;
+import io.vertx.ext.couchdb.CouchdbClientOptions;
 import io.vertx.ext.web.client.HttpRequest;
 import io.vertx.ext.web.client.WebClient;
-import io.vertx.ext.web.client.WebClientOptions;
 
 public class CouchdbClientImpl implements CouchdbClient {
 
-  public static CouchdbClient create(Vertx vertx, WebClientOptions options) {
-    return new CouchdbClientImpl(vertx, options);
+  public static CouchdbClient create(Vertx vertx, CouchdbClientOptions clientOptions) {
+    return new CouchdbClientImpl(vertx, clientOptions);
   }
 
-  private final Vertx vertx;
-  private final WebClientOptions options;
+  private final VertxInternal vertx;
+  private final CouchdbClientOptions clientOptions;
   private WebClient client = null;
-  private Credentials credentials;
 
-  CouchdbClientImpl(Vertx vertx, WebClientOptions options) {
-    this.vertx = vertx;
-    this.options = options;
+  CouchdbClientImpl(Vertx vertx, CouchdbClientOptions clientOptions) {
+    this.vertx = (VertxInternal) vertx;
+    this.clientOptions = clientOptions;
   }
 
   @Override
   public Future<JsonObject> status() {
-
     Promise<JsonObject> promise = Promise.promise();
     initClient();
     HttpRequest<?> request = client.get("/");
 
-    if (this.credentials != null) {
-      request.authentication(this.credentials);
+    if (clientOptions.getCredentials() != null) {
+      request.authentication(clientOptions.getCredentials().toAuthCredentials());
```

Comment:
We can write that smarter:

`public class AnonymousCrerdentials implements Credentials`

see: https://vertx.io/docs/apidocs/io/vertx/ext/auth/authentication/Credentials.html

then 

```java
Credentials getCredentials() {
return clientOptions.getCredentials() != null ? clientOptions.getCredentials() : new AnonymousCrerdentials();
}

```

Reply:
Got it. I'll check and refactor it